compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEDistributeCoresAndObjectFifos.cpp

```diff
@@ -56,182 +61,238 @@ struct LocationMapInfo {
+                        rewriter.create<vector::TransferReadOp>(
+                            transferReadOp.getLoc(), transferReadOp.getType(),
+                            newAlloc, newIndices,
+                            transferReadOp.getPermutationMapAttr(),
+                            transferReadOp.getPadding(),
+                            transferReadOp.getMask(),
+                            transferReadOp.getInBoundsAttr());
+                    rewriter.replaceAllUsesWith(transferReadOp,
+                                                newTransferReadOp.getResult());
+                    toBeErased.push_back(transferReadOp);
+                    return success();
+                  })
+              .Case<vector::TransferWriteOp>(
+                  [&](vector::TransferWriteOp transferWriteOp) {
+                    rewriter.setInsertionPoint(transferWriteOp);
+                    // Since in this function we're basically changing the L1
+                    // sizes of the Alloc, for dimensions with size as 1 we need
+                    // to set the indices as 0. We need to do this at this step
+                    // because there would be loop dependencies on the same and
+                    // when we unroll those loops later in this pass we would
+                    // have incorrect offset values being formed for those
+                    // dimension.
+                    SmallVector<Value> newIndices =
+                        transferWriteOp.getIndices();
+                    Value c0 = rewriter.create<arith::ConstantIndexOp>(
+                        transferWriteOp.getLoc(), 0);
+
+                    for (unsigned i = 0, n = newShape.size(); i < n; i++) {
+                      if (newShape[i] == 1) newIndices[i] = c0;
+                    }
+
+                    rewriter.create<vector::TransferWriteOp>(
+                        transferWriteOp.getLoc(), transferWriteOp.getVector(),
+                        newAlloc, newIndices,
+                        transferWriteOp.getPermutationMapAttr(),
+                        transferWriteOp.getMask(),
+                        transferWriteOp.getInBoundsAttr());
+                    toBeErased.push_back(transferWriteOp);
+                    return success();
+                  })
+              .Case<memref::ExtractStridedMetadataOp>(
+                  [&](memref::ExtractStridedMetadataOp
+                          extractStridedMetadataOp) {
+                    rewriter.setInsertionPoint(extractStridedMetadataOp);
+                    auto newextractStridedMetadataOp =
+                        rewriter.create<memref::ExtractStridedMetadataOp>(
+                            extractStridedMetadataOp.getLoc(), newAlloc);
+                    rewriter.replaceAllUsesWith(
+                        extractStridedMetadataOp.getResults(),
+                        newextractStridedMetadataOp.getResults());
+                    toBeErased.push_back(extractStridedMetadataOp);
```

Comment:
It seems like not much is happening here and this could be written as a more general handler for different (future) ops? (Just a thought, not necessarily for this PR)

Reply:
Ok. I'll refactor in a later PR. This part of the change was moving an if-else statement into an llvm switch statement.